libs/api/domains/application/src/lib/application.resolver.ts

```diff
@@ -142,4 +144,23 @@ export class ApplicationResolver {
       user.authorization,
     )
   }
+
+  @Mutation(() => Application, { nullable: true })
+  requestFileSignature(
+    @Args('input') input: RequestFileSignatureInput,
+    @CurrentUser() user: User,
+  ): Promise<Application> {
+    return this.applicationService.requestFileSignature(
+      input,
+      user.authorization,
+    )
+  }
+
+  @Mutation(() => Application, { nullable: true })
+  uploadSignedFile(
+    @Args('input') input: UploadSignedFileInput,
+    @CurrentUser() user: User,
+  ): Promise<Application> {
+    return this.applicationService.uploadSignedFile(input, user.authorization)
+  }
```

Comment:
Where will this mutation be triggered? Shouldn't the user _just_ start the signing process, and when that's done, we somehow upload the file automatically?

Reply:
Yes this flow is weird. So currently it is like this:
1. User requests for file signature. We send back a control code (pin) which is displayed to the user and a document token. 
2. When the user receives these items we send another request which calls the signing service with that document token to check if the signature was successful. In the meantime the user approves/rejects electronic signature on their phone.
It would be ideal that this was one call from the client but since we have to return the control code and then know if the file upload was successful in the client then I implemented this as two calls. This is the same way as it is implemented in the judicial system. 
We could go for server sent event here. But not sure how that works with a graphql layer or if that is a overkill. Any suggestions/thoughts?

Reply:
Ok, thanks for the explanation. I think we can revisit this later.